plugin/evm/testutils/metrics.go

```diff
@@ -0,0 +1,21 @@
+package testutils
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/metrics"
+)
+
+// WithMetrics enables go-ethereum metrics globally for the test.
+// If the [metrics.Enabled] is already true, nothing is done.
+// Otherwise, it is set to true and is reverted to false when the test finishes.
+func WithMetrics(t *testing.T) {
+	panicIfCallsFromNonTest()
```

Comment:
I wonder if we should rather do that with a CI script/linter. Not sure if this is maintainable for all functions that we want to avoid using in prod code.